services/wallet/cmd/rest_run.go

```diff
@@ -21,12 +22,16 @@ import (
 // This function takes a cobra command and starts up the
 // wallets rest microservice.
 func WalletRestRun(command *cobra.Command, args []string) {
-	// setup generic middlewares and routes for health-check and metrics
-	r := cmd.SetupRouter(command.Context())
-	r, ctx, _ := wallet.SetupService(command.Context(), r)
+	ctx, service := wallet.SetupService(command.Context())
+	router := cmd.SetupRouter(command.Context())
```

Comment:
should we be passing in the context that we got from SetupService to SetupRouter?